python/activator/middleware_interface.py

```diff
@@ -160,6 +188,15 @@ def ingest_image(self, oid: str) -> None:
         # subprocess.run(cmd, check=True)
 
     def run_pipeline(self, visit: Visit, snaps) -> None:
+        """Process the received image.
+
+        Parameters
+        ----------
+        visit : Visit
+            The visit information of the image to be processed.
```

Comment:
Again, a `Visit` is not a visit.

Reply:
Missing docstring for `filter_calibs`, below.

Reply:
I'm only going to document the things as we start to use them: this only appears in commented-out code.